src/wp-includes/post.php

```diff
@@ -6481,6 +6493,22 @@ function wp_delete_attachment_files( $post_id, $meta, $backup_sizes, $file ) {
 		}
 	}
 
+	// Check for alternate full size mime types in the root sources array to delete.
+	if ( isset( $meta['sources'] ) && is_array( $meta['sources'] ) ) {
+		$sources = $meta['sources'];
+		array_shift( $sources );
+		foreach ( $sources as $mime => $properties ) {
+			if ( ! is_array( $properties ) || empty( $properties['file'] ) ) {
+				continue;
+			}
+
+			$intermediate_file = str_replace( wp_basename( $file ), $properties['file'], $file );
+			if ( ! wp_delete_file_from_directory( $intermediate_file, $intermediate_dir ) ) {
+				$deleted = false;
+			}
```

Comment:
Shouldn't these files be deleted in the same way as the main `$file`? Would this work as simply the following?
```suggestion
			if ( ! wp_delete_file_from_directory( $properties['file'], $uploadpath['basedir'] ) ) {
				$deleted = false;
			}
```